[distributed][ez] use gloo enum instead of hardcoding string

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27652 [distributed][ez] use gloo enum instead of hardcoding string**

Changes "gloo" to dist.backend.GLOO in `rpc_test.py`. Closes #27231

Test plan:
`python test/test_rpc_fork.py -v && python test/test_rpc_spawn.py -v`

Differential Revision: [D17845067](https://our.internmc.facebook.com/intern/diff/D17845067/)

rohan-varma committed:
[pytorch][ez] use gloo enum instead of hardcoding stirng

Changes "gloo" to dist.backend.GLOO in rpc_test.py.

Differential Revision: [D17845067](https://our.internmc.facebook.com/intern/diff/D17845067/)

[ghstack-poisoned]

xush6528 reviewed (approved):
Thanks for improving maintainability

rohan-varma committed:
Update on "[distributed][ez] use gloo enum instead of hardcoding string"


Changes "gloo" to dist.backend.GLOO in `rpc_test.py`. Closes #27231

Test plan:
`python test/test_rpc_fork.py -v && python test/test_rpc_spawn.py -v`

Differential Revision: [D17845067](https://our.internmc.facebook.com/intern/diff/D17845067/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed][ez] use gloo enum instead of hardcoding string"


Changes "gloo" to dist.backend.GLOO in `rpc_test.py`. Closes #27231

Test plan:
`python test/test_rpc_fork.py -v && python test/test_rpc_spawn.py -v`

Differential Revision: [D17845067](https://our.internmc.facebook.com/intern/diff/D17845067/)

[ghstack-poisoned]

rohan-varma commented:
failures are due to existing issues
